File: Jongryul/07/16194.py
import sys
input = sys.stdin.readline

n = int(input())
arr = [0] + list(map(int, input().split()))
dp = [arr[i] for i in range(n+1)]

for i in range(1, n+1):
    for j in range(1, i):
        if dp[i] > dp[i-j] + dp[j]:
            dp[i] = dp[i-j] + dp[j]
print(dp[n])

File: Jongryul/07/16194.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const n = Number(input[0]);
const arr = [...[0], ...input[1].split(" ").map(Number)];
const dp = Array.from({ length: n + 1 }, (_, i) => arr[i]);

for (let i = 0; i < n + 1; i++) {
  for (let j = 0; j < i; j++) {
    if (dp[i] > dp[i - j] + dp[j]) {
      dp[i] = dp[i - j] + dp[j];
    }
  }
}

console.log(dp[n]);
